src/System.CommandLine/IDirectiveCollection.cs

```diff
@@ -2,13 +2,14 @@
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
 using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
 
 namespace System.CommandLine
 {
     public interface IDirectiveCollection : IEnumerable<KeyValuePair<string, IEnumerable<string>>>
     {
         bool Contains(string name);
 
-        bool TryGetValues(string name, out IEnumerable<string> values);
+        bool TryGetValues(string name,  [NotNullWhen(true)] out IEnumerable<string>? values);
```

Comment:
nit: inconsistent formatting of attribute here and elsewhere. Not sure what is the guideline for this repo. For Roslyn, we're moving to having a space after `]`.